src/mindlessgen/molecules/generate_molecule.py

```diff
@@ -135,6 +142,22 @@ def generate_atom_list(cfg: GenerateConfig, verbosity: int = 1) -> np.ndarray:
                     + "Setting the number of atoms for the fixed composition. "
                     + f"Returning: \n{natoms}\n"
                 )
+            # If the molecular charge is defined, and a fixed element composition is defined, check if the electrons are even. If not raise an error.
+            if cfg.molecular_charge:
+                protons = calculate_protons(natoms)
+                nel = protons - cfg.molecular_charge
+                f_elem = any(
+                    count > 0 and (i in get_lanthanides() or i in get_actinides())
+                    for i, count in enumerate(natoms)
+                )
+                if (f_elem and calculate_ligand_protons(natoms, nel) % 2 != 0) or (
```

Comment:
What does `calculate_ligand_protons` lead to when no `f` elements are present? Would be not good if an error can occur in this setting.

Reply:
Sure, the condition would never be true. I just wanted to make sure that no `RuntimeError` or similar can occur when executing `calculate_ligand_electrons` although there are no `f_elem`s. But if that does work, it's fine.